packages/amplify-provider-awscloudformation/src/override-manager/amplify-factory-transform.ts

```diff
@@ -0,0 +1,17 @@
+import { AmplifyCategories, CLISubCommands, IAmplifyResource } from 'amplify-cli-core';
+import { AmplifyAuthTransform } from '@aws-amplify/amplify-category-auth';
+import { AmplifyCategoryTransform } from 'amplify-category-plugin-interface';
+import { AmplifyUserPoolGroupTransform } from '@aws-amplify/amplify-category-auth/lib/provider-utils/awscloudformation/auth-stack-builder';
+
+export class AmplifyCategoryTransformFactory {
+  public static getCategoryTransformInstance(resource: IAmplifyResource): AmplifyCategoryTransform[] {
+    if (resource.category === AmplifyCategories.AUTH) {
+      return [
+        new AmplifyUserPoolGroupTransform(resource.resourceName, CLISubCommands.UPDATE),
+        new AmplifyAuthTransform(resource.resourceName, CLISubCommands.UPDATE),
+      ];
+    } else if (resource.category === AmplifyCategories.STORAGE) {
```

Comment:
can we make this factory/utility generic so that we don't have to add if statements for all categories?